src/she/sdl2/sdl2_display.cpp

```diff
@@ -0,0 +1,363 @@
+  {
+    return false;
+
+    if (cursor == m_nativeCursor) {
+      return true;
+    }
+
+    if (m_cursor)
+      SDL_FreeCursor(m_cursor);
+    m_cursor = SDL_DISABLE;
+
+    switch (cursor) {
+    case kNoCursor: break;
+    case kArrowCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_ARROW); break;
+    case kIBeamCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_IBEAM); break;
+    case kWaitCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_WAIT); break;
+    // case kHelpCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HELP); break;
+    case kForbiddenCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_NO); break;
+    case kMoveCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HAND); break;
+    // case kLinkCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_LINK); break;
+    case kSizeNSCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENS); break;
+    case kSizeWECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEWE); break;
+    // case kSizeNCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEN); break;
+    // case kSizeNECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENE); break;
+    // case kSizeECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEE); break;
+    // case kSizeSECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZESE); break;
+    // case kSizeSCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZES); break;
+    // case kSizeSWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZESW); break;
+    // case kSizeWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEW); break;
+    // case kSizeNWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENW); break;
+    default: break;
+    }
+
+    SDL_SetCursor(m_cursor);
+    m_nativeCursor = cursor;
+    return m_cursor ? true : false;
+  }
+
+  void SDL2Display::setMousePosition(const gfx::Point& position)
+  {
+    SDL_WarpMouseInWindow(
+      m_window,
+      m_scale * position.x,
+      m_scale * position.y);
+  }
+
+  void SDL2Display::captureMouse()
+  {
+    SDL_CaptureMouse(SDL_TRUE);
+  }
+
+  void SDL2Display::releaseMouse()
+  {
+    SDL_CaptureMouse(SDL_FALSE);
+  }
+
+  std::string SDL2Display::getLayout()
+  {
+// #ifdef _WIN32
+//   WINDOWPLACEMENT wp;
+//   wp.length = sizeof(WINDOWPLACEMENT);
+//   if (GetWindowPlacement((HWND)nativeHandle(), &wp)) {
+//     std::ostringstream s;
+//     s << 1 << ' '
+//       << wp.flags << ' '
+//       << wp.showCmd << ' '
+//       << wp.ptMinPosition.x << ' '
+//       << wp.ptMinPosition.y << ' '
+//       << wp.ptMaxPosition.x << ' '
+//       << wp.ptMaxPosition.y << ' '
+//       << wp.rcNormalPosition.left << ' '
+//       << wp.rcNormalPosition.top << ' '
+//       << wp.rcNormalPosition.right << ' '
+//       << wp.rcNormalPosition.bottom;
+//     return s.str();
+//   }
+// #endif
```

Comment:
We should probably implement this in a cross-platform way and remove this comment.

Reply:
Do we actually even need it?

Reply:
It's used to remember the last size and position of the window to restore it the next time you open it. I think this functionality is already built in to Windows itself, so we might not need it. It only ever worked on Windows anyways.

Reply:
I guess it is best to leave window sizes to the OS to remember. So these comments should be removed and these functions should just be no-ops.

Reply:
I guess we can replicate some of it with `SDL_GetWindowPosition`, `SDL_GetWindowSize`.